discord_rate_limiter.py

```
@@ -0,0 +1,185 @@
+        """
+        max_retries = 5
+        base_delay = 1.0
+        
+        for attempt in range(max_retries):
+            try:
+                # Check for global rate limit
+                if self._global_rate_limit_reset:
+                    current_time = time.time()
+                    if current_time < self._global_rate_limit_reset:
+                        sleep_time = self._global_rate_limit_reset - current_time
+                        logger.warning(f"Global rate limit active, sleeping for {sleep_time:.2f} seconds")
```

Pylint (3.3.7)

[convention] Line too long (106/100)

(C0301)